### PR TITLE
solveLCPmex now checks for gurobi at run time

### LINKED ISSUE #934: solveLCPmex causes failures on minimal builds
Site: DRAKE003
Build Name: precompiled-RobotLocomotion-master
Build Server Directory: C:/cygwin64/tmp/drake-precompiled-32559/drake
CDash: http://kobol.csail.mit.edu/cd/viewTest.php?onlyfailed&buildid=12201

Most recent commits to drake included in this build:
```
fac5ce2 Merge pull request #926 from
avalenzu/invalid-time-MinDistanceConstraint
2aa5177 Merge pull request #925 from
avalenzu/hotfix-drakeCollision-memory-leak
f6bb2f1 Merge pull request #924 from tkoolen/fix-testDrakeGradientUtil
4cc8d4c Merge pull request #908 from psiorx/jc-solvelcpmex-fastqp
e9fdc81 Correctly handle invalid time in C++ MinDistanceConstraint
```

370 tests run:
261 passed
72 cancelled (with missing dependencies  SDPSolver:9, avl:4, bullet:11,
gloptipoly3:3, gurobi:8, iris:3, lcm:26, mosek:2, octomap:1, sedumi:3,
yalmip:2)
0 timed out
37 failed
 4 <a href="https://github.com/RobotLocomotion/drake/issues/374">known
issue #374</a>: failing imu unit tests
 1 <a href="https://github.com/RobotLocomotion/drake/issues/466">known
issue #466</a>: implement Jdot*v for drakeFunction.kinematic methods
 6 <a href="https://github.com/RobotLocomotion/drake/issues/569">known
issue #569</a>: lcppath.mexw64 and lcppath.mexw32 fails to load on win
machines
 1 <a href="https://github.com/RobotLocomotion/drake/issues/585">known
issue #585</a>: testIndividualCentersOfPressure fails
 25 unknown

solvers/trajectoryOptimization/test/contactImplicitBrick (unknown issue)
```
error_id = Simulink:SFunctions:SFcnErrorStatus
error_message = Error reported by S-function 'DCSFunction' in
'TimeSteppingRigidBodyManipulator_063595169242585000/DrakeSys':
0
is_segfault = 0
```
```Matlab
C:/Program Files/MATLAB/R2012a/bin/win64/MATLAB.exe

To get started, type one of these: helpwin, helpdesk, or demo.
For product information, visit www.mathworks.com.

To reproduce this test use rng(817844152,'twister')

{Warning: Disabling autosave for simulink blocks (to avoid generating a
lot of *.mdl.autosave files in your
directory.  If you aren't a regular Simulink user and don't want this
disabled, comment out this section
in addpath_drake.
To disable this warning in the future, add
  warning('off','Drake:DisablingSimulinkAutosave')
to your matlab startup.m} 
{> In addpath_drake at 74
 In addpath_drake at 4} 
Calling addpath_snopt
DrakeSystem S-Function: error when calling ''update'' with the following
arguments:
 TimeSteppingRigidBodyManipulator

 Properties:
                     timestep: 0.1250
                         twoD: 0
             position_control: 0
                    LCP_cache: [1x1 SharedDataHandle]
                enable_fastqp: 0
   lcmgl_contact_forces_scale: 0
         z_inactive_guess_tol: 0.0100
                         umin: [0x1 double]
                         umax: [0x1 double]
              warning_manager: [1x1 WarningManager]


    0

        0
        0
   0.8000
   0.0917
  -0.1129
   0.0431
        0
        0
        0
        0
        0
        0

Undefined function 'setupLCPmex' for input arguments of type
'DrakeMexPointer'.

Error in TimeSteppingRigidBodyManipulator/solveMexLCP (line 241)
       [z, Mqdn, wqdn] = setupLCPmex(obj.manip.mex_model_ptr, q, v, u,
phiC, n, D, obj.timestep, obj.z_inactive_guess_tol, obj.LCP_cache.data.z);

Error in TimeSteppingRigidBodyManipulator/update (line 187)
          [obj,z,Mqdn,wqdn] = solveMexLCP(obj,t,x,u);

Error in DynamicalSystem/simulate (line 110)
simout = sim(mdl,pstruct);

Error in DrakeSystem/simulate (line 490)
       [varargout{:}] = simulate@DynamicalSystem(obj,varargin{:});

Error in contactImplicitBrick (line 20)
xtraj_ts = simulate(plant_ts,[0 tf],x0);

Error in fevalPackageSafe (line 21)
 feval(f);
Error using DynamicalSystem/simulate (line 110)
Error reported by S-function 'DCSFunction' in
'TimeSteppingRigidBodyManipulator_063595169242585000/DrakeSys':
0

Error in DrakeSystem/simulate (line 490)
       [varargout{:}] = simulate@DynamicalSystem(obj,varargin{:});

Error in contactImplicitBrick (line 20)
xtraj_ts = simulate(plant_ts,[0 tf],x0);

Error in fevalPackageSafe (line 21)
 feval(f);


<test_name>solvers/trajectoryOptimization/test/contactImplicitBrick</test_name> <error_id>Simulink:SFunctions:SFcnErrorStatus</error_id> <error_message>Error reported by S-function 'DCSFunction' in 'TimeSteppingRigidBodyManipulator_063595169242585000/DrakeSys':
0</error_message> 


```

### PR DESCRIPTION
**Pull request by psiorx**
...and caches the result during compile.

Resolves #934

### CODE REVIEW

**Comment by RussTedrake** [comment]
looks good.  just waiting for the build servers.

File: systems/plants/TimeSteppingRigidBodyManipulator.m [hunk 1]
```diff
@@ -118,6 +119,7 @@ function checkDirty(obj)
 
     function model = compile(model)
       w = warning('off','Drake:RigidBodyManipulator:UnsupportedContactPoints');
+      model.gurobi_present = checkDependency('gurobi');
```
Comment (rdeits) [inline systems/plants/TimeSteppingRigidBodyManipulator.m]: With this commit, we get the `l_init_called failed` error on the Atlas computers every time we try to start up the controller (specifically, every time we try to enter `constructQPDataPointerMex`. Switching it to `model.gurobi_present = true` prevents that error from occurring.

Comment (rdeits) [inline systems/plants/TimeSteppingRigidBodyManipulator.m]: This call to gurobi is how we derived the minimal crash discussed here: https://github.com/mitdrc/drc/pull/1251#issuecomment-89026122

Comment (psiorx) [inline systems/plants/TimeSteppingRigidBodyManipulator.m]: Also worth noting is that inside the checkDependency implementation, commenting out the try/catch block that actually calls gurobi to check for a license stopped the segfault.

Comment (RussTedrake) [inline systems/plants/TimeSteppingRigidBodyManipulator.m]: Sounds like a red herring.  Just moved memory and the other bug reappeared..?

- Russ

**Pull request merged**